lib/heap_sort.rb

```diff
@@ -1,8 +1,24 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity:  O(n log n)
+# Space Complexity: O(1) / O(n) 1 because the heap is being sorted in place but I'm creating a new array to return the answer, which will be n elements long
 def heap_sort(list)
```

Comment:
You're building a heap as an external data structure, so the space complexity is O(n).  You CAN do this with O(1) space complexity but you need to use `list` as the internal structure of the array.